pimd: eBGP integration for SA loop detection

rzalamena opened the pull request:
This PR implements the optional knob to enable eBGP integration to detect loops in SAs. The detection of MSDP SA loops works as follow:
1. eBGP AS number is configured in the MSDP peer
2. When SA is received the RP value is used in RPF check
2.a. If the RPF route AS Path number matches the neighbor AS then the SA is accepted
2.b. if the RPF route AS Path number doesn't match the neighbor (or doesn't exist) the SA is rejected (means that the SA is not coming from the shortest path)

donaldsharp commented:
Is there any particular reason we've added a second slightly different address lookup for pim instead of just modifying the first one?

rzalamena commented:
> Is there any particular reason we've added a second slightly different address lookup for pim instead of just modifying the first one?

Removed the new Zebra API and incremented the existing one (zebra next hop lookup). Now there is just a slight variation on the next hop lookup in PIM: there is a new function that does next hop lookup and also returns the BGP AS number.

riw777 reviewed (commented):
code looks okay to me ... it would be good to have someone with more pom code exposure look at this before pushing ...

what happens here if the AS Path changes once PIM has "read" it?

donaldsharp commented:
I do not see documentation for the new clear command?  It would be nice that it was tested as well.  Once that happens I'll get it in.
